homeassistant/components/siren/__init__.py

```diff
@@ -51,14 +51,24 @@ class SirenTurnOnServiceParameters(TypedDict, total=False):
     volume_level: float
 
 
-def filter_turn_on_params(
+def process_turn_on_params(
     siren: SirenEntity, params: SirenTurnOnServiceParameters
 ) -> SirenTurnOnServiceParameters:
-    """Filter out params not supported by the siren."""
+    """
+    Process turn_on service params.
+
+    Filters out unsupported params and validates the rest.
+    """
     supported_features = siren.supported_features or 0
 
     if not supported_features & SUPPORT_TONES:
         params.pop(ATTR_TONE, None)
+    elif ATTR_TONE in params and (
+        not siren.available_tones
```

Comment:
If this short-circuits, there will be an error below since the walrus expression won't have been evaluated.

Reply:
This also means we're missing a test for this case.

Reply:
Wonder if we should raise a RuntimeError if the entity is implemented with support for tone but without available tones? I think that's a developer error.

Reply:
good points, another PR incoming

Reply:
> Wonder if we should raise a RuntimeError if the entity is implemented with support for tone but without available tones? I think that's a developer error.

I don't fully agree with that. Just because tones aren't available now, doesn't mean it doesn't support it? Maybe they do need to be added by the user? E.g., upload sounds or activate something?

Maybe I'm just overthinking it though.

Reply:
Yeah, ok.